sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/global/ThroughputControlContainerManager.java

```diff
@@ -0,0 +1,185 @@
+            })
+            .retryWhen(RetrySpec.max(1).filter(throwable -> {
+                CosmosException cosmosException = Utils.as(Exceptions.unwrap(throwable), CosmosException.class);
+                return cosmosException != null && cosmosException.getStatusCode() == HttpConstants.StatusCodes.CONFLICT;
+            }))
+            .flatMap(itemResponse -> {
+                this.configItem = itemResponse.getItem();
+
+                if (!expectedConfigItem.equals(configItem)) {
+                    logger.warn(
+                        "Group config using by this client is different than the one in control container, will be ignored. Using following config: {}" +
+                            "targetThroughput: {}, targetThroughputThreshold: {}",
+                        this.configItem.toString());
+                }
+
+                return Mono.just(this.configItem);
+            });
+    }
+
+    /**
+     * Query the load factor of all clients for the group.
+     *
+     * @return The sum of load factor from all clients.
+     */
+    public Mono<Double> queryLoadFactorFromAllClients() {
+        // The current design is using ttl to expire client items, so there is no need to check whether the client item is expired.
+        return this.globalControlContainer.readAllItems(new PartitionKey(this.clientItemPartitionKeyValue), ThroughputGlobalControlClientItem.class)
+            .collectList()
+            .flatMapMany(clientItemList -> Flux.fromIterable(clientItemList))
+            .map(clientItem -> clientItem.getLoadFactor())
+            .reduce((x1, x2) -> x1 + x2 );
+    }
+
+    /**
+     * Update the existing group client item.
+     *
+     * If resource not found, then create a new client item.
+     * The client item may get deleted based on the ttl if the client can not keep updating the item due to unexpected failure (for example, network failure).
+     *
+     * @param loadFactor The new load factor of the client.
+     * @return A {@link ThroughputGlobalControlClientItem};
+     */
+    public Mono<ThroughputGlobalControlClientItem> replaceOrCreateGroupClientItem(double loadFactor) {
+        return Mono.just(this.clientItem)
+            .flatMap(groupClientItem -> {
+                groupClientItem.setLoadFactor(loadFactor);
+                return this.globalControlContainer.replaceItem(groupClientItem, groupClientItem.getId(), new PartitionKey(groupClientItem.getGroup()));
+            })
+            .onErrorResume(throwable -> {
+                CosmosException cosmosException = Utils.as(Exceptions.unwrap(throwable), CosmosException.class);
+                if (cosmosException != null && cosmosException.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
+                    logger.warn("Can not find the expected client item {}, will recreate a new one", this.clientItem.getId());
+                    return this.globalControlContainer.createItem(this.clientItem);
```

Comment:
409 dance missing

Reply:
Since there will only ever be one client updating the corresponding client item, so it should be safe to take itself as the source of truth. So do not really think a need for 409 specific handling here.

 Will change to blind retry 5 times in case any exception for the create item exception.